src/app/public/src/modules/shared/route-metadata.service.ts

```diff
@@ -18,12 +18,17 @@ export class StacheRouteMetadataService {
   constructor(
     @Inject(STACHE_ROUTE_METADATA_SERVICE_CONFIG)
     public metadata: any[]) {
-      this.metadata.forEach(route => {
-        if (route.order) {
-          route.order = numberConverter(route.order);
-        }
-      });
+      this.metadata.forEach((route: any) => this.validateNavOrder(route));
     }
+
+  private validateNavOrder(route: any): void {
+    if (route.hasOwnProperty('order')) {
+      route.order = numberConverter(route.order);
+      if (route.order <= 0 || route.order !== parseInt(route.order, 10)) {
```

Comment:
I learned something new today! Would you mind putting the logic in a variable for the sake of readability?

Also, aren't we supporting zero? I think we should.

```
const isDecimal = (route.order !== parseInt(route.order, 10));
if (route.order < 0 || isDecimal) {
  delete route.order;
}
```